configuration_about_project.txt
Instructions for Peregrine FYP:

=================================================

1- Install docker and docker-compose
2- run the command "docker-compose up -d" at same directory where the docker-compose.yml is placed.

Note: Make sure that all ports 3000, 8000 and 3306 are empty.
Incase: If the port is occupied of DB change the port with "3307:3306" in docker-compose.yml's DB section.

3- Now for IPv4 we have to do following steps( This is for providing the backend with the ip of docker-network of app):
 
 -> docker network ls
 -> docker network inspect <docker-network-id>
 
NOTE: This ID will be the project name ID

4- In Container Section you will find out IPv4 of DB container.

5- There will be an IP as IPv4, Just copy that IP at some place (without /16,/32,/8,/24)


================= To Configure DB ================

following commands for configuration:


- docker ps
Copy the DB container ID

- docker cp Peregrine.sql <conatiner_id>:.
- docker exec -ti <container_id> bash
- ls 


NOTE: "ls" to confirm that the .sql is there or not




- mysql -u root -p Peregrine_Falcon < Peregrine.sql

#Password = Rawim1989 

- mysql -u root -p

- SHOW DATABASES;
- use Peregrine_Falcon;
- SHOW TABLES;
- select * from auth_user;

============== To configure copied IP in backend ==========

1- Replace the copied IP in setting.py and SAVE file. (Main Project)
2- Follow commands to delete the container of backend (in order to re-build):

- docker ps 

copy the container_id of backend

- docker rm -f <container_id>
- docker images

find the backend image

- docker rmi -f <image_id>

after that again

- docker-compose up -d


** start the app with this url : http://localhost:3000/














**** In case of containers running :

List all running containers:

- docker ps


Stop any running containers that are using the image:

docker stop [container_name/container_id]

Replace [container_name/container_id] with the name or ID of the container.

Remove all stopped containers that are using the image:

docker rm [container_name/container_id]

Again, replace [container_name/container_id] with the name or ID of the container.

Finally, remove the image:

docker rmi -f [image_name/image_id]

Replace [image_name/image_id] with the name or ID of the image.

peregrine_project/settings.py
"""
Django settings for peregrine_project project.

Generated by 'django-admin startproject' using Django 4.1.7.

For more information on this file, see
https://docs.djangoproject.com/en/4.1/topics/settings/

For the full list of settings and their values, see
https://docs.djangoproject.com/en/4.1/ref/settings/
"""

from pathlib import Path
from datetime import timedelta

# Build paths inside the project like this: BASE_DIR / 'subdir'.
BASE_DIR = Path(__file__).resolve().parent.parent


# Quick-start development settings - unsuitable for production
# See https://docs.djangoproject.com/en/4.1/howto/deployment/checklist/

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'django-insecure-a-%jd!5b&q5+u=cg1mu1b-uuhqrrpk3n3m96@n10dr%ef#37=#'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = []


# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'peregrine_app',
    'rest_framework',
    'rest_framework.authtoken',
    'rest_framework_simplejwt',
    'peregrine_app.peregrine_api',
    'rest_framework_simplejwt.token_blacklist',
    'corsheaders',
    'peregrine_app.facade_testing',
]

MIDDLEWARE = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'corsheaders.middleware.CorsMiddleware',
]

ROOT_URLCONF = 'peregrine_project.urls'

TEMPLATES = [
    {
        'BACKEND': 'django.template.backends.django.DjangoTemplates',
        'DIRS': [],
        'APP_DIRS': True,
        'OPTIONS': {
            'context_processors': [
                'django.template.context_processors.debug',
                'django.template.context_processors.request',
                'django.contrib.auth.context_processors.auth',
                'django.contrib.messages.context_processors.messages',
            ],
        },
    },
]



WSGI_APPLICATION = 'peregrine_project.wsgi.application'


# Database
# https://docs.djangoproject.com/en/4.1/ref/settings/#databases

DATABASES = {  
    'default': {  
        'ENGINE': 'django.db.backends.mysql',  
        'NAME': 'Peregrine_Falcon',  
        'USER': 'root',  
        'PASSWORD': 'Rawim1989',  
        'HOST': '172.20.0.2',  
        'PORT': '3306',  
        'OPTIONS': {  
            'init_command': "SET sql_mode='STRICT_TRANS_TABLES'"  
        },
    'test': {
        'ENGINE': 'django.db.backends.sqlite3',
        'NAME': ':memory:',
    }  
    }  
}


# Password validation
# https://docs.djangoproject.com/en/4.1/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]


# Internationalization
# https://docs.djangoproject.com/en/4.1/topics/i18n/

LANGUAGE_CODE = 'en-us'

TIME_ZONE = 'UTC'

USE_I18N = True

USE_TZ = True


# Static files (CSS, JavaScript, Images)
# https://docs.djangoproject.com/en/4.1/howto/static-files/

STATIC_URL = 'static/'

# Default primary key field type
# https://docs.djangoproject.com/en/4.1/ref/settings/#default-auto-field

DEFAULT_AUTO_FIELD = 'django.db.models.BigAutoField'

# AUTHENTICATION_BACKENDS = [
#     'peregrine_app.backend.UserrBackend',
#     'django.contrib.auth.backends.ModelBackend',
# ]


REST_FRAMEWORK = {
    'DEFAULT_AUTHENTICATION_CLASSES': [
        'rest_framework_simplejwt.authentication.JWTAuthentication',
        'rest_framework.authentication.TokenAuthentication'
    ],
}


SIMPLE_JWT = {
    'ACCESS_TOKEN_LIFETIME': timedelta(minutes=180),
    'REFRESH_TOKEN_LIFETIME': timedelta(days=1),
    'ROTATE_REFRESH_TOKENS': True,
    'BLACKLIST_AFTER_ROTATION': True,
    'ALGORITHM': 'HS256',
    'SIGNING_KEY': SECRET_KEY,
    'VERIFYING_KEY': None,
    'AUTH_HEADER_TYPES': ('Bearer',),
    'USER_ID_FIELD': 'id',
    'USER_ID_CLAIM': 'user_id',
    'AUTH_TOKEN_CLASSES': ('rest_framework_simplejwt.tokens.AccessToken',),
    'TOKEN_TYPE_CLAIM': 'token_type',
}



# This logging setting is configured in order not to document each time the server restarts after every change
LOGGING = {
    'version': 1,
    'disable_existing_loggers': False,
    'handlers': {
        'console': {
            'class': 'logging.StreamHandler',
        },

    },
    'loggers': {
        'django.utils.autoreload': {
            'handlers': ['console'],
            'propagate': False,
        },

    },
}

# Allowing requests from REACT
CORS_ORIGIN_WHITELIST = [    'http://localhost:3000',]
